### PR TITLE
Loki 43b: Add S3 image storage, remove in-DB storage

### PR DESCRIPTION
**Pull request by jessetaylornci**
This pull request moves the image storage from the database itself to cloud-stored in S3 buckets.

### CODE REVIEW

File: backend/src/main/java/com/nci/loki/admin/services/impl/S3InteractionServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+			e.printStackTrace();
+		}
+		
+		s3Key.setKey(fileNameKey);
+		return s3Key;
+	}
+	
+	private String generateFileNameKey(Customer customer) {
+		return "image-upload-" + customer.getCustomerId() + "-" + System.currentTimeMillis();
+	}
```
Comment (rfxu) [inline backend/src/main/java/com/nci/loki/admin/services/impl/S3InteractionServiceImpl.java]: Use a constant for text string.  Could the user double click on the upload button?

File: backend/src/main/java/com/nci/loki/admin/services/impl/S3InteractionServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		try {
+			InputStream imageInputStream = image.getInputStream();
+			s3Client.putObject(new PutObjectRequest(S3_BUCKET, fileNameKey, imageInputStream, metaData));
+			LOG.info("Successfully saved image from CustomerId {} using UniqueCode {}",
+					customer.getCustomerId(), customer.getUniqueCode());
+			imageInputStream.close();
+		} catch (IOException e) {
+			LOG.info("Could not convert file to input stream for code " + customer.getUniqueCode());
+			e.printStackTrace();
+		}
```
Comment (rfxu) [inline backend/src/main/java/com/nci/loki/admin/services/impl/S3InteractionServiceImpl.java]: Should use LOG.error instead of printStack

Comment (rfxu) [reply backend/src/main/java/com/nci/loki/admin/services/impl/S3InteractionServiceImpl.java]: If there is a error does the imageInputStream need to be closed in a final?

Comment (jessetaylornci) [reply backend/src/main/java/com/nci/loki/admin/services/impl/S3InteractionServiceImpl.java]: .close() itself requires a try-catch block, so it is best to keep it in the try{} block.

**Pull request merged**